nncf/experimental/torch/nas/bootstrapNAS/search/search.py

```diff
@@ -193,19 +210,33 @@ def __init__(self,
         self._verbose = verbose
         self._top1_accuracy_validation_fn = None
         self._val_loader = None
-        evo_algo = search_config['algorithm']
-        if evo_algo == EvolutionaryAlgorithms.NSGA2.value:
+        self._algorithm_name = search_config['algorithm']
+        sampling = get_sampling("int_lhs")
+        if self._algorithm_name == EvolutionaryAlgorithms.NSGA2.value:
```

Comment:
I'd expect that search algorithm implements a single algorithm – a single responsibility. Otherwise, it would hard to support a lot of if-conditions.

Reply:
Fixed. Thanks!